probe/kubernetes/controls.go

```diff
@@ -80,6 +173,44 @@ func (r *Reporter) deleteVolumeSnapshot(req xfer.Request, namespaceID, volumeSna
 	}
 }
 
+// Describe will parse the nodeID and return response according to the node (resource) type.
+func (r *Reporter) Describe() func(xfer.Request) xfer.Response {
+	return func(req xfer.Request) xfer.Response {
+		var f func(req xfer.Request) xfer.Response
+		_, tag, ok := report.ParseNodeID(req.NodeID)
+		if !ok {
+			return xfer.ResponseErrorf("Invalid ID: %s", req.NodeID)
+		}
+		switch tag {
+		case "<pod>":
+			f = r.CapturePod(r.describePod)
+		case "<service>":
+			f = r.CaptureService(r.describeService)
```

Comment:
There seems to be a lot of repetition going on here - could it be done in a data-driven way?
E.g. a table containing function pointers like `ParseServiceNodeID`, `WalkServices`?

Reply:
Thanks for the review @bboreham 
Will update the PR soon.

Reply:
> How did we aquire a dependency on vendor/k8s.io/cli-runtime/pkg/genericclioptions ?

k8s.io/kubernetes/pkg/kubectl/describe/interface.go requires the `genericclioptions` package. Because of this we acquired this dependency.